### PR TITLE
My plan: Move Jetpack plan thank you to modal

### PR DESCRIPTION
**Pull request by sirreal**
Closes https://github.com/Automattic/wp-calypso/issues/33319

Moves overlay obscuring card into modal.

**This is a first pass largely to invite design feedback.**

### After

Screen Shot 2019-05-29 at 11 27 02

### Before

Screen Shot 2019-05-29 at 11 26 17

## Testing instructions

* Purchase a plan for a Jetpack site
* Alternatively just got to `/plans/my-plan/SITE_SLUG?thank-you` for a connected Jetpack site.
* Modal works? Dismisses properly?

### CODE REVIEW

File: client/my-sites/plans/current-plan/index.jsx [hunk 1]
```diff
@@ -131,7 +123,11 @@ class CurrentPlan extends Component {
 				<QuerySitePlans siteId={ selectedSiteId } />
 				{ shouldQuerySiteDomains && <QuerySiteDomains siteId={ selectedSiteId } /> }
 
-				{ ! showThankYou && <PlansNavigation path={ path } /> }
+				<Dialog isVisible={ showThankYou }>
+					{ isFreePlan ? <FreePlanThankYouCard /> : <PaidPlanThankYouCard /> }{' '}
```
Comment (simison) [inline client/my-sites/plans/current-plan/index.jsx]: Why the `{' '}`?

**Comment by sirreal** [comment]
> From a design perspective, it looks like we have some excessive margins around the card which throws the card a little off balance.

Thanks @fditrapani! I'd simply moved the `Card` into `Dialog`, both added padding. I've cleaned that up.

I've also removed some margins that become unnecessary in the `Dialog`. Here are some screens (different color scheme is unrelated):

mobile

tablet

Slightly more padding on `>800px` viewport width:
desk

**Comment by fditrapani** [comment]
That looks great! thank you.

**Comment by sirreal** [comment]
As I reworked this, we stopped dealing with `Card`s and names stopped making sense. That lead to violating lint rules about class name conventions, which lead to a significant directory restructuring. That lead to discovering this was part of the old CSS monolith, so I did a quick grep and moved the dialog styles to the new pipeling with a sass import from the component.

I ended up dropping the extra padding on desktop, the additional 8px were barely noticeable (that ok, @fditrapani?).

This could use a quick look over of the code and smoke test. Thanks!

**Comment by fditrapani** [comment]
Looks good to me @sirreal!

File: client/my-sites/plans/current-plan/index.jsx (hunk 1 above)
Comment (sirreal) [reply client/my-sites/plans/current-plan/index.jsx]: That was unintended. Removed in 6c600afffe7b7e9fa5b27c5b69140df25658772f. Thanks for spotting!

File: client/my-sites/plans/current-plan/index.jsx [hunk 2]
```diff
@@ -131,7 +123,11 @@ class CurrentPlan extends Component {
 				<QuerySitePlans siteId={ selectedSiteId } />
 				{ shouldQuerySiteDomains && <QuerySiteDomains siteId={ selectedSiteId } /> }
 
-				{ ! showThankYou && <PlansNavigation path={ path } /> }
+				<Dialog isVisible={ showThankYou }>
```
Comment (simison) [inline client/my-sites/plans/current-plan/index.jsx]: I've sometimes noticed we show "free plan thank you" before the paid plan kicks in. Should we just add `!isRequestingSitePlans` check here? Totally not blocking and we can loop back to this later.

Comment (simison) [reply client/my-sites/plans/current-plan/index.jsx]: Actually there already seems to be `this.isLoading` helper for that.

**Pull request merged**